extensions/hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/CsvInputFormat.java

```diff
@@ -110,19 +148,32 @@ public NullWritable getCurrentKey() {
             }
 
             @Override
-            public Object getCurrentValue() {
-                return current;
+            public Object getCurrentValue() throws IOException {
+                String current = reader.getCurrentValue().toString();
+                if (formatClazz.equals(String[].class)) {
```

Comment:
nit: IIRC you can simply do formatClazz == String[].class

Reply:
Done in ec0def77